### PR TITLE
Put deleted status message on InfrastructureReady condition

### PR DESCRIPTION
**Pull request by thedadams**
A previous change mistakenly set the Provision status when a machine was
being deleted. This change now updates the InfrastructureReady condition
when it has been detected that the infrastructure object has been 
deleted.

This is necessary because the CAPI machine controller is waiting for the
infrastructure object to be deleted and this is required so that the 
CAPI controller runs again.

Issues:
https://github.com/rancher/rancher/issues/34235
https://github.com/rancher/rancher/issues/34264

### CODE REVIEW

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Review by paynejacob — DISMISSED** [review]

**Comment by dramich** [comment]
Why not keep the `Deleting` and update the massage to indicate the machine part. This feels heavy and not totally accurate since the object does still exist.

**Review by dramich — COMMENTED** [review]
Per offline discussion, ensure we don't change the status/message/condition of the machine if it is being deleted since it's already in the correct state. Revert the previous fix, and find a location to enqueue the machine again after we have done our work on delete.

**Comment by thedadams** [comment]
@dramich I updated the description to try to indicate why these changes are necessary.

**Review by snasovich — DISMISSED** [review]
Code LGTM, leaving it to @dramich to confirm logic is correct.

**Review by paynejacob — APPROVED** [review]

File: pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go [hunk 1]
```diff
@@ -32,7 +32,8 @@ import (
 )
 
 const (
-	Provisioned = condition.Cond("Provisioned")
+	Provisioned         = condition.Cond("Provisioned")
+	InfrastructureReady = condition.Cond("InfrastructureReady")
```
Comment (dramich) [inline pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go:36]: Per offline, attempt to use the upstream CAPI consts here

File: pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go [hunk 2]
```diff
@@ -266,7 +275,7 @@ func (h *handler) getInfraMachineState(capiMachine *capi.Machine) (status corev1
 	machine, err := h.dynamic.Get(gvk, capiMachine.Namespace, capiMachine.Spec.InfrastructureRef.Name)
 	if apierror.IsNotFound(err) {
 		if capiMachine.DeletionTimestamp != nil {
-			return corev1.ConditionUnknown, "MachineDeleted", "machine is being deleted", "", nil
+			return corev1.ConditionFalse, "Deleted", "machine infrastructure is deleted", "", err
```
Comment (dramich) [inline pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go:278]: Per offline, attempt to use the upstream CAPI consts here

File: pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go (hunk 1 above)
Comment (thedadams) [reply pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go:36]: There is a Condition constant for `InfrastructureReady`, but not for `Provisioned`.

File: pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go (hunk 2 above)
Comment (thedadams) [reply pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go:278]: Will use the `MachineDelete` reason constant.

File: pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go [hunk 3]
```diff
@@ -163,6 +164,15 @@ func (h *handler) OnChange(key string, machine *capi.Machine) (*capi.Machine, er
 		return machine, nil
 	}
 
+	status, reason, message, providerID, err := h.getInfraMachineState(machine)
+	if machine.DeletionTimestamp != nil && apierror.IsNotFound(err) {
+		// If the machine is being deleted and the  infrastructure machine object is not found,
```
[the remark this answers is not in the document]
Comment (thedadams) [reply pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go:169]: Done.

**Review by dramich — APPROVED** [review]

**Pull request merged**